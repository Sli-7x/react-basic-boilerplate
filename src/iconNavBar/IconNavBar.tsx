import * as React from 'react';
import { Icon } from '../index';
import { Container, Item } from './IconNavBar.style';

export interface IconNavBarItem {
  id: string;
  title: string;
  icon: string;
}

export interface IKeys {
  id: string;
  title: string;
  icon: string;
}

export interface IIconNavBarProps {
  list: IconNavBarItem[];
  value: string | number;
  name: string;
  keys?: IKeys;
  onSelect: (id: string, name: string) => void;
}

export class IconNavBar extends React.Component<IIconNavBarProps, any> {
  constructor(props: IIconNavBarProps) {
    super(props);
    this.onClick = this.onClick.bind(this);

    this.state = {
      keys: props.keys ? props.keys : { id: 'id', title: 'title', childs: 'childs' },
    };
  }

  onClick(id: string) {
    this.props.onSelect(id, this.props.name);
  }

  render() {
    const { keys } = this.state;
    const { value } = this.props;

    return (
      <Container>
        {this.props.list.map((val: any, i: number) => (
          <Item key={i} selected={value === val[keys.id]}>
            <Icon onClick={this.onClick.bind(this, val[keys.id])} name={val[keys.icon]} />
          </Item>
        ))}
      </Container>
    );
  }
}
