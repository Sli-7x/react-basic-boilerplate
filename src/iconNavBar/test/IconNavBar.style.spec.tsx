import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Item } from '../IconNavBar.style';

describe('IconNavBar.style', () => {
  describe('Item', () => {
    const props = {
      selected: false,
    };

    it('should render correctly Item', () => {
      const tree = renderer.create(<Item {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should cover when passed selected=true to Item', () => {
      const tree = renderer.create(<Item selected={true} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
