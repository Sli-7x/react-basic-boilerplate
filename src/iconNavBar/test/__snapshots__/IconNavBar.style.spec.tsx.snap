// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconNavBar.style Item should cover when passed selected=true to Item 1`] = `
<div
  className="sc-bwzfXH bmWnPJ"
  selected={true}
/>
`;

exports[`IconNavBar.style Item should render correctly Item 1`] = `
<div
  className="sc-bwzfXH eqnpGN"
  selected={false}
/>
`;
